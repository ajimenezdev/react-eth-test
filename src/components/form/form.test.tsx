import React from "react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import { act } from "@testing-library/react-hooks";
import { fireEvent, render, waitForElement } from "@testing-library/react";

import Form from "./index";

it("renders without crashing when no arguments", () => {
  const component = renderer.create(<Form onUpdateSearch={() => null} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("call onUpdateSearch when correct values", async () => {
  const mockCallback = jest.fn();
  const { getByLabelText, getAllByRole, container } = render(
    <Form onUpdateSearch={mockCallback} />
  );

  const address = getByLabelText("address");
  const network = getByLabelText("select_network");
  act(() => {
    fireEvent.change(address, {
      target: { value: "0xc350de406f54271c0e025bf72855418d65f8bcbc" }
    });
  });
  const selectButton = getAllByRole("button")[0];
  act(() => {
    fireEvent.click(selectButton);
  });
  await waitForElement(() => getByLabelText("select_network_mainnent"), {
    container
  });
  const mainnet = getByLabelText("select_network_mainnent");
  act(() => {
    fireEvent.click(mainnet);
  });

  act(() => {
    fireEvent.click(getByLabelText("search"));
  });

  expect(mockCallback.mock.calls.length).toBe(1);
});

it("do not call onUpdateSearch when wrong eth address", async () => {
  const mockCallback = jest.fn();
  const { getByLabelText, getAllByRole, container } = render(
    <Form onUpdateSearch={mockCallback} />
  );

  const address = getByLabelText("address");
  const network = getByLabelText("select_network");
  fireEvent.change(address, {
    target: { value: "wrong_address" }
  });
  const selectButton = getAllByRole("button")[0];
  fireEvent.click(selectButton);
  await waitForElement(() => getByLabelText("select_network_mainnent"), {
    container
  });
  const mainnet = getByLabelText("select_network_mainnent");
  fireEvent.click(mainnet);

  await fireEvent.click(getByLabelText("search"));

  expect(mockCallback.mock.calls.length).toBe(0);
});

it("do not call onUpdateSearch when empty", () => {
  const mockCallback = jest.fn();
  const { getByLabelText } = render(<Form onUpdateSearch={mockCallback} />);

  fireEvent.click(getByLabelText("search"));

  expect(mockCallback.mock.calls.length).toBe(0);
});
