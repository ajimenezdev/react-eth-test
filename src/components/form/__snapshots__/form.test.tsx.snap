// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing when no arguments 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 body-form MuiPaper-rounded"
>
  <form
    className="form"
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
      >
        Address
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          aria-label="address"
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
      </div>
      
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
      >
        Select Network
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          aria-label="select_network"
          type="hidden"
          value=""
        />
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
      
    </div>
    <button
      aria-label="search"
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Search
      </span>
    </button>
  </form>
</div>
`;
